wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -2076,6 +2233,178 @@ private function translate_regexp_functions( $token ) {
 		return true;
 	}
 
+	/**
+	 * Detect GROUP BY.
+	 *
+	 * @param WP_SQLite_Token $token The token to translate.
+	 *
+	 * @return bool
+	 */
+	private function capture_group_by( $token ) {
+		if (
+			! $token->matches(
+				WP_SQLite_Token::TYPE_KEYWORD,
+				WP_SQLite_Token::FLAG_KEYWORD_RESERVED,
+				array( 'GROUP' )
+			)
+		) {
+			return false;
+		}
+		$next = $this->rewriter->peek_nth( 2 )->value;
+		if ( 'BY' !== strtoupper( $next ) ) {
+			return false;
+		}
+
+		$this->has_group_by = true;
```

Comment:
Would `$this->last_reserved_keyword` or something to that effect work? This here is only meant to have impact on the very next `HAVING` clause, but in this current form a nested query that has a `GROUP BY` would trigger a false-positive for the outer query that has an ungrouped `HAVING` query.

Reply:
Man, that's an edge case. Some plugin author's writing SQL code with subqueries AND misuse of HAVING, they can doggone well fix it. The testcase for this came from Advanced Database Cleaner, without subqueries. I'll contact the author.